build.settings.ps1

```diff
@@ -55,7 +55,24 @@ Properties {
     [System.Diagnostics.CodeAnalysis.SuppressMessage('PSUseDeclaredVarsMoreThanAssigments', '', Scope='*', Target='CertSubjectName')]
     $CertSubjectName = $null
 
-
+    # -------------------- Script Analysis properties ------------------------------
+
+    # The script analysis task step will run, unless your host is in the array defined below.
+    # This allows you to control whether code analysis is executed, for hosts where script
+    # analysis is included in the product.
+    [System.Diagnostics.CodeAnalysis.SuppressMessage('PSUseDeclaredVarsMoreThanAssigments', '', Scope='*', Target='SkipCodeAnalysisHost')]
+    $SkipCodeAnalysisHost = @(
+        'Visual Studio Code Host',
+        'My Custom Host with scriptanalyzer support'
+    )
+
+    # To control the failure of the build with specific script analyzer rule severities,
+    # the CodeAnalysisStop variable can be used. The supported values for this variable are
+    # 'Warning', 'Error', 'All', 'None' or 'Skip'. Invalid input will stop on all rules.
+    # 'Skip' will skip over the code analysis step all together.
+    [System.Diagnostics.CodeAnalysis.SuppressMessage('PSUseDeclaredVarsMoreThanAssigments', '', Scope='*', Target='CodeAnalysisStop')]
+    $CodeAnalysisStop = 'Error'
```

Comment:
Can we rename this to "ScriptAnalysisAction"?  That would we similar to the VS Code Analysis ruleset where you can select the Action per rule.   Ditto for SkipCodeAnalysisHost -> SkipScriptAnalysisHost.  Just to keep it more in the "PowerShell vernacular".